DbTables/inc/TrkElementStatus.hh

```diff
@@ -0,0 +1,93 @@
+//
+// tables for tracker element status.  An
+// element is a straw, panel, or plane
+//  These tables list just defects, so are variable length
+//  Original author: Dave Brown (LBNL)
+
+#ifndef DbTables_TrkElementStatus_hh
+#define DbTables_TrkElementStatus_hh
+
+#include <string>
+#include <iomanip>
+#include <sstream>
+#include <map>
+#include <regex>
+#include "DataProducts/inc/StrawId.hh"
+#include "DataProducts/inc/StrawIdMask.hh"
+#include "DataProducts/inc/StrawStatus.hh"
+#include "DbTables/inc/DbTable.hh"
+
+namespace mu2e {
+
+  class TrkElementStatus : public DbTable {
+  public:
+
+    // define a struct for each table row
+    struct TrkElementStatusRow {
+      StrawId _sid;
+      StrawStatus _status;
+      StrawId const& id() const { return _sid; }
+      StrawStatus const& status() const { return _status; }
+      TrkElementStatusRow(StrawId const& sid, StrawStatus const& status) : _sid(sid), _status(status) {}
+    };
+
+    typedef std::shared_ptr<TrkElementStatus> ptr_t;
+    typedef std::shared_ptr<const TrkElementStatus> cptr_t;
+
+    TrkElementStatus(const char* Name, const char* DbName, StrawIdMask sidmask, StrawStatus statusmask):
+      DbTable(Name,DbName, "strawid,strawstatus"), _sidmask(sidmask), _statusmask(statusmask) {}
+    const TrkElementStatusRow& rowAt(const std::size_t index) const { return _rows.at(index);}
+    std::vector<TrkElementStatusRow> const& rows() const {return _rows;}
+    std::size_t nrow() const override { return _rows.size(); };
+    // this is a variable-size table, so don't overrido nrowFix()
+    size_t size() const override { return baseSize() + nrow()*sizeof(TrkElementStatusRow); }
+    // table-specific info
+    StrawIdMask const& sidMask() const { return _sidmask; }
+    StrawStatus const& statusMask() const { return _statusmask; }
```

Comment:
statusMask doesn't seem to be used anywhere, maybe I missed it.

Reply:
Thanks for catching that, I just forgot to add the check.  Done now.